src/main/java/core/basesyntax/service/impl/FruitTransactionsHandlerImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionHandler;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.BalanceHandler;
+import core.basesyntax.strategy.impl.PurchaseHandler;
+import core.basesyntax.strategy.impl.ReturnHandler;
+import core.basesyntax.strategy.impl.SupplyHandler;
+import java.util.List;
+
+public class FruitTransactionsHandlerImpl implements FruitTransactionHandler {
+    private final FruitStorage fruitStorage;
+    private OperationStrategy operationStrategy = new OperationStrategy(
+            new BalanceHandler(),
+            new PurchaseHandler(),
+            new ReturnHandler(),
+            new SupplyHandler());
+
+    public FruitTransactionsHandlerImpl(FruitStorage fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
```

Comment:
Can't see reasons not to use unified approach to dependencies
```suggestion
    private OperationStrategy operationStrategy;
    
    public FruitTransactionsHandlerImpl(FruitStorage fruitStorage, OperationStrategy operationStrategy) {
        this.fruitStorage = fruitStorage;
        this.operationStrategy = operationStrategy;
    }
```